src/filesystem.ts

```diff
@@ -105,11 +106,32 @@ export async function loadFileSystem(
 /**
  * Create a new filesystem and assign it to a user.
  *
- * Warning: This function will override a user's filesystem with an empty one.
- *
  * @param permissions The permissions to initialize the filesystem
+ * @param options Optional params
+ * @param options.reset Override an existing filesystem with an empty one
  */
-export const bootstrapFileSystem = async (permissions: Permissions): Promise<FileSystem> => {
+export const bootstrapFileSystem = async (
+  permissions: Permissions,
+  options: { reset?: boolean } = { reset: false }
+): Promise<FileSystem> => {
+  const { reset } = options
+  const authedUsername = await common.authenticatedUsername()
+
+  // Check for authed user and existing filesystem
+  if (authedUsername) {
+    if (!reset) {
+      const dataCid = navigator.onLine ? await dataRoot.lookup(authedUsername) : null // data root on server or DNS
+      const logCid = await cidLog.newest() // data root in browser
+
+      if (dataCid !== null || logCid !== undefined) {
+        throw new Error("Bootstrap operation will reset an existing filesystem. Please set reset to true if this behavior is desired.")
```

Comment:
Maybe put 'reset' in quotation marks or something like that.

Reply:
Sounds good, will do. 👌